apps/server/src/services/ElasticSearchClient.ts

```diff
@@ -342,31 +370,62 @@ export async function deleteRemovedInstitutions(
     return;
   }
   info("deleting institutions", deletedInstitutionsIds);
-  const deletePromises = deletedInstitutionsIds.map(async (ucpId: string) => {
-    return await ElasticsearchClient.delete({
-      index: "institutions",
-      id: ucpId,
-    });
-  });
 
   try {
-    await Promise.all(deletePromises);
+    if (config.ELASTIC_SEARCH_SINGLE_THREAD) {
+      for (const ucpId of deletedInstitutionsIds) {
+        await ElasticsearchClient.delete({
+          index: "institutions",
+          id: ucpId,
+        });
+        console.info(`Deleted institution ${ucpId}`);
+      }
+    } else {
+      await Promise.all(
+        deletedInstitutionsIds.map(async (ucpId: string) => {
```

Comment:
Looks like you could share this function and use deletedInstitutionsIds.forEach for one a Promise.all for the other

Reply:
Same thing for update and index

Reply:
Ah actually never mind. Can't await with forEach. Should probably create a function for the duplicated code though.